timeseries/src/autogluon/timeseries/models/local/statsforecast.py

```diff
@@ -0,0 +1,252 @@
+import logging
+from typing import List, Type, Union
+
+import pandas as pd
+
+from autogluon.core.utils.exceptions import TimeLimitExceeded
+from autogluon.timeseries.dataset.ts_dataframe import ITEMID, TIMESTAMP, TimeSeriesDataFrame
+from autogluon.timeseries.models.local.abstract_local_model import AbstractLocalModel
+from autogluon.timeseries.utils.hashing import hash_ts_dataframe_items
+
+logger = logging.getLogger(__name__)
+
+
+class AbstractStatsForecastModel(AbstractLocalModel):
+    """Wrapper for StatsForecast models.
+
+    Cached predictions are stored inside the model to speed up validation & ensemble training downstream.
+
+    Attributes
+    ----------
+    allowed_local_model_args : List[str]
+        List of allowed arguments that can be passed to the underlying model.
+        Arguments not in this list will be filtered out and not passed to the underlying model.
+    """
+
+    allowed_local_model_args: List[str] = []
+    DEFAULT_N_JOBS: Union[float, int] = -1
+
+    def get_model_type(self) -> Type:
+        raise NotImplementedError
+
+    def _fit(self, train_data, time_limit=None, verbosity=2, **kwargs) -> None:
+        """Prepare hyperparameters that will be passed to the underlying model.
+
+        As for all local models, actual fitting + predictions are delegated to the ``predict`` method.
+        """
+        # TODO: Find a way to ensure that SF models respect time_limit
+        # Fitting usually takes >= 20 seconds
+        if time_limit is not None and time_limit < 20:
+            raise TimeLimitExceeded
+        super()._fit(train_data=train_data, time_limit=time_limit, verbosity=verbosity, **kwargs)
```

Comment:
One quick idea to achieve `time_limit` is to take advantage of `timeout` from `ThreadPoolExecutor` or `ProcessPoolExecutor`. Details can be found [here](https://docs.python.org/3/library/concurrent.futures.html)

Reply:
I agree but, if I understand correctly, this would require monkey-patching StatsForecast. Is it fine if we leave it as-is for now and add a `TODO` comment?

Reply:
Let's add it in TODO and follow up with that